Schedule

cmochsner opened the pull request:
Schedule and view my rides component and associated service. Added links to navbar. Add tripUri to environment files. Updated environment files for production and dev mode.

cmochsner committed:
Created schedule and view my rides components.

cmochsner committed:
Merge branch 'dev' of https://github.com/Batch-738-Mathew-Oberlies/RideshareFrontend into dev

cmochsner committed:
Co-authored-by: Quinn Bierbaum<quinn.bierbaum@gmail.com>
Added schedule and view my rides components with associated service. Updated environment files for tripUri.

cmochsner committed:
Readded tripUri to environment.ts

Quinn-Patrick committed:
Merge branch 'dev' into schedule

IkenoXamos commented on src/app/components/schedule/schedule.component.ts:
```diff
@@ -0,0 +1,61 @@
+      }
+      );
+  }
+  addTrip(t: Trip)
```
For readability, please have empty lines between method declarations, and perhaps some js-doc comments explaining what they are doing,

IkenoXamos commented on src/app/components/schedule/schedule.component.ts:
```diff
@@ -0,0 +1,61 @@
+
+       this.serv.getRiderTrips(this.id).subscribe(
+        data =>{ this.riderTrips=data
+      }
```
These curly brackets aren't necessary. Can be structured similarly to the getTrips() method.

IkenoXamos commented on src/app/components/view-my-rides/view-my-rides.component.html:
```diff
@@ -0,0 +1,64 @@
+<div class="container centerMe">
+    <div class="card mt-2"> 
+        <table class="table table-hover">
```
Work on reducing code duplication by either creating a shared component for the table, or just one component for the entire view that can be reused in both places.

IkenoXamos commented on src/app/components/view-my-rides/view-my-rides.component.ts:
```diff
@@ -0,0 +1,51 @@
+      data => this.trips=data
+       );
+    }
+  }
```
Please add empty lines between methods for readability.

IkenoXamos commented on src/app/services/schedule-service/schedule.service.ts:
```diff
@@ -0,0 +1,36 @@
+  constructor(private http: HttpClient) 
+  { 
+  }
+  getTrips(): Observable<Trip[]>
```
Please add empty lines between methods for readability.

IkenoXamos reviewed (changes requested):
Are any of the changes to all of the 3 environment files necessary for this PR?

Quinn-Patrick committed:
Fixed small concerns, the "remove curly brackets" issue became more
complicated since it became part of a bug fix. Christina is still
working on reducing code duplication.

Quinn-Patrick committed:
Merge branch 'schedule' of https://github.com/Batch-738-Mathew-Oberlies/RideshareFrontend into schedule

cmochsner committed:
Abstracted common html code into one file

cmochsner committed:
Merge branch 'schedule' of https://github.com/Batch-738-Mathew-Oberlies/RideshareFrontend into schedule

cmochsner committed:
deleted unused imports

cmochsner committed:
cleaned up unnecessary code in environment files

cmochsner committed:
Deleted leftover view-my-ride component html file

cmochsner committed:
fixed code smells

cmochsner commented:
We added a new endpoint (tripUri), which is necessary. Other changes are not neccessary but were made to delete unneeded code. I believe the Google API is now being used on the backend and is no longer needed on the frontend.

IkenoXamos commented on src/app/components/schedule/schedule.component.ts:
```diff
@@ -0,0 +1,73 @@
+  public id: number;
+  
+  constructor(private serv: ScheduleService, private auth: AuthService) 
+  { 
```
For consistency across the codebase, please move opening curly brackets to the same line as the method declaration.

cmochsner committed:
reverted environment files to original format, aside from the addition of tripUri

cmochsner committed:
fixed line for readability
